frontend/ui/widget/checkbutton.lua

```diff
@@ -0,0 +1,119 @@
+--[[--
+Button widget that shows a checkmark (`✓`) when checked and an empty box (`□`)
+when unchecked.
+
+Example:
+
+    local CheckButton = require("ui/widget/CheckButton")
+    local parent_widget = OverlapGroup:new{}
+    table.insert(parent_widget, CheckButton:new{
+        text = _("Show password"),
+        callback = function() end,
+    })
+    UIManager:show(parent_widget)
+
+]]
+
+local Device = require("device")
+local Font = require("ui/font")
+local GestureRange = require("ui/gesturerange")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local TextWidget = require("ui/widget/textwidget")
+local UIManager = require("ui/uimanager")
+local Screen = Device.screen
+
+local CheckButton = InputContainer:new{
+    callback = nil,
+    hold_callback = nil,
+    checked = false,
+    enabled = true,
+    overlap_align = "right",
+    text = nil,
+    toggle_text = nil,
+    window = nil,
+
+    padding = Screen:scaleBySize(5),
+    margin = Screen:scaleBySize(5),
+    bordersize = Screen:scaleBySize(3),
+}
+
+function CheckButton:init()
+    self:initCheckButton(self.checked)
+end
+
+function CheckButton:initCheckButton(checked)
+    self.checked = checked
+    self[1] = TextWidget:new{
+        text = (self.checked and "✓ " or "□ ") .. (self.text or ""),
```

Comment:
Perhaps. I actually wanted to use ☐ and ☑ (matched in size and everything) but our GUI font doesn't have that character. I'm not really sure how appropriate the square root is regardless. If anything I'm included to change *that*. :-P

http://www.fileformat.info/info/unicode/char/2610/index.htm
http://www.fileformat.info/info/unicode/char/2611/index.htm

Of course the consistent width thing is a non-issue except in this first draft. I'll make it a separate widget so that the width for one of them can also used on the other.